### PR TITLE
WIP Preparing a documentation endpoint served in deck.

### PR DESCRIPTION
**Pull request by droslean**
ref #5279

### CODE REVIEW

File: prow/cmd/deck/static/prism.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/* PrismJS 1.15.0
+https://prismjs.com/download.html#themes=prism-coy&languages=bash+git+yaml+vim&plugins=line-numbers+toolbar+highlight-keywords+command-line+copy-to-clipboard */
```
Comment (BenTheElder) [inline prow/cmd/deck/static/prism.js]: License? looks like this is MIT which requires including the license https://github.com/PrismJS/prism/blob/master/LICENSE

File: prow/cmd/deck/static/docs.html [hunk 2]
```diff
@@ -0,0 +1,304 @@
+<!DOCTYPE html>
```
Comment (BenTheElder) [inline prow/cmd/deck/static/docs.html]: how is this file managed?

File: prow/cmd/deck/static/docs.html [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+                    </a>Add more jobs by modifying <strong>config.yaml</strong></h2>
+                    <p>Create a file called <strong>config.yaml</strong>, and add the following to it:</p>
+
+                    <pre class="line-numbers"><code class="language-yaml">periodics:
+                                - interval: 10m
+                                  agent: kubernetes
+                                  name: echo-test
+                                  spec:
+                                  containers:
+                                   &nbsp - image: alpine
```
Comment (BenTheElder) [inline prow/cmd/deck/static/docs.html]: nit: entities should be like `&nbsp;`

**Comment by stevekuznetsov** [comment]
I think we should start with more of a design-focused discussion here. I think the goals for the framework are:

 - let Prow document itself on it's own front-end
 - let Prow documentation pull from the running Prow instance wherever possible so that documentation is up-to-date

The goals for the doc are:
 - document how to configure the cluster (we have enough information to show people where the config files live by parsing the plugins config)
 - document other Prow-related workflows
 - ???

We should hash those out first before continuing to an implementation.

File: prow/cmd/deck/static/docs.html (hunk 3 above)
Comment (droslean) [reply prow/cmd/deck/static/docs.html]: there is some issue with the yaml formatting. I am working on it.

**Comment by droslean** [comment]
We decide that this is not a good time for PR until we will finalize the documentation itself. However, most probably the docs will be hosted in different endpoints and not under the deck.
So, I am closing the PR.

**Pull request closed without merging**